### PR TITLE
feat: project structure redesign and more features

### PR DESCRIPTION
**Pull request by e-cloud**
* split up documentation in README.md into `/docs`
* redesign directory structure for better code management
* use logging module to enable rotating logs, for better problem tracing
* improve application bootstrap flow

### CODE REVIEW

File: application.py [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
+
+    with open('logging-conf.yaml', 'r') as f:
+        config = yaml.safe_load(f.read())
+        logging.config.dictConfig(config)
+
+    from app import create_app
+
+    app = create_app()
+
+    app.run(host='127.0.0.1', port=port)
```
Comment (kunkun-tang) [inline application.py]: I referred to [this skeleton](https://github.com/aws-samples/eb-python-flask/blob/master/application.py) here, so I'm not sure if this change breaks. I do not entirely understand the diff between 0.0.0.0 and 127.0.0.1  Do you?

File: application.py [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
     )
     args = parser.parse_args()
     port = args.port
 
-    application.run(host='0.0.0.0', port=port)
+    import logging.config
+    import yaml
+    import os
+
+    if not os.path.exists('./logs'):
```
Comment (kunkun-tang) [inline application.py]: Not sure whether this path breaks eb (elastic beanstalk) environment. May we use WORKING_FOLDER as the prefix path?

File: application.py [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
+        os.makedirs('./logs')
+
+    with open('logging-conf.yaml', 'r') as f:
+        config = yaml.safe_load(f.read())
+        logging.config.dictConfig(config)
```
Comment (kunkun-tang) [inline application.py]: I already see an `import logging.config` in this file.  Should we just use config directly?

File: app/services/tencent/send_sms.py [hunk 4]
```diff
@@ -33,9 +36,9 @@ def send_message(phone, random_num):
             86, phone_numbers[0], template_id, params, sign=sms_sign, extend='', ext=''
         )  # 签名参数未提供或者为空时，会使用默认签名发送短信
     except HTTPError as e:
-        print(e)
+        log.debug(e)
```
Comment (kunkun-tang) [inline app/services/tencent/send_sms.py]: The logging feature is a great plus! Thanks!

File: app/__init__.py [hunk 5]
```diff
@@ -1,51 +1,29 @@
...
+def create_app(**kwargs):
+    """
+    Entry point to the Flask RESTful Server application
+    ===================================================
+    """
+    # This is a workaround for Alpine Linux (musl libc) quirk:
+    # https://github.com/docker-library/python/issues/211
+    import threading
+
+    threading.stack_size(2 * 1024 * 1024)
```
Comment (kunkun-tang) [inline app/__init__.py]: This change is beneficial to docker only? are you using docker as development environment? @e-cloud

Comment (kunkun-tang) [reply app/__init__.py]: And why 2M, this number?

File: app/extensions/__init__.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+
+logging = Logging()
+
+cross_origin_resource_sharing = CORS(supports_credentials=True)
+
+db = SQLAlchemy()
+
+force_auto_coercion()
+force_instant_defaults()
```
Comment (kunkun-tang) [inline app/extensions/__init__.py]: googled but still don't get it how `force_auto_coercion` works out.

File: app/resources/__init__.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+=========
+
+Resources enable logical resource separation.
+
+You may control enabled resources by modifying ``ENABLED_RESOURCES`` config
+variable.
+"""
+import logging
+
+log = logging.getLogger('resources')
```
Comment (kunkun-tang) [inline app/resources/__init__.py]: what is resources here for? why don't use __name__

File: application.py (hunk 1 above)
Comment (e-cloud) [reply application.py]: 0.0.0.0 is different in windows.

File: application.py (hunk 2 above)
Comment (e-cloud) [reply application.py]: `WORKING_FOLDER` is just a custom environment, why would it have special effect?

File: application.py (hunk 3 above)
Comment (e-cloud) [reply application.py]: I am not sure what you mean. `import logging.config` is just a module import.

File: app/__init__.py (hunk 5 above)
Comment (e-cloud) [reply app/__init__.py]: no, just some copy pasted code from reference library. I could disable it.

File: app/extensions/__init__.py (hunk 6 above)
Comment (e-cloud) [reply app/extensions/__init__.py]: another copy-pasted code. I believe it have some value. https://sqlalchemy-utils.readthedocs.io/en/latest/listeners.html From the docs, it says will automatically cast data base on it specified type from SQLAlchemy's mapper.

**Comment by e-cloud** [comment]
@kunkun-tang plz review again

File: application.py (hunk 1 above)
Comment (kunkun-tang) [reply application.py]: Seems 0.0.0.0 is a must. https://stackoverflow.com/a/52185250

File: application.py (hunk 2 above)
Comment (kunkun-tang) [reply application.py]: I'm not sure how and where EB launches the python process. Is it possible `.logs` will be placed under a different folder?

File: application.py (hunk 3 above)
Comment (kunkun-tang) [reply application.py]: I'm not certain on python syntax. If that's the case, should we change the name? Say you created a variable called `config`, and import a module with the same name (logging.confg).

Comment (e-cloud) [reply application.py]: don't worry, python will handle it correctly.

**Comment by e-cloud** [comment]
@kunkun-tang please have a review again. I've change some code to support dynamic host ip, logs dir, and working folder.

**Comment by e-cloud** [comment]
@kunkun-tang please have a review again. I've change some code to support dynamic host ip, logs dir, and working folder.

**Pull request merged**